### ISSUE TITLE
Popover: nested popovers don't open

### ISSUE THREAD

**Issue by raduflp** [tailwindlabs/headlessui#2285]
**What package within Headless UI are you using?**
@headlessui/react

**What version of that package are you using?**
1.7.8 - 1.7.10

**What browser are you using?**
Firefox, Chrome, Safari

**Reproduction URL**
https://stackblitz.com/edit/react-ts-tqzsk8
for the working version change the version of @headlessui/react in `package.json` to 1.7.7 

**Describe your issue**
Starting with `@headlessui/react v1.7.8` you cannot open a Popover from an existing `Popover.Panel`. It worked fine in 1.7.7.
I've attached a minimal example. Just for context, in our scenario we have a drill down UX where the user can navigate a few levels deep.

**Comment by laneme** [comment]
Popovers doesnt open even when not nested. Might wanna edit your title.

For those who cant get it to work even after downgrading to `1.7.7`,  might wanna check your lock file. Doing this to a fixed version should solve it.
```diff
-"@headlessui/react": "^1.7.7",
+"@headlessui/react": "1.7.7",
```

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

This should be fixed by #2293, and will be available in the next release.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.
